src/fold/events/__init__.py

```diff
@@ -0,0 +1,109 @@
+# Copyright (c) 2022 - Present Myalo UG (haftungbeschränkt) (Mark Aron Szulyovszky, Daniel Szemerey) <info@dreamfaster.ai>. All rights reserved. See LICENSE in root folder.
+from __future__ import annotations
+
+from typing import Callable, List, Optional, Tuple
+
+import pandas as pd
+
+from ..base import Artifact, Composite, Pipeline, Pipelines, T, Transformation, fit_noop
+from ..utils.list import wrap_in_list
+from .base import EventFilter, Labeler
+from .filters import EveryNth, NoFilter
+from .labeling import BinarizeFixedForwardHorizon
+
+
+def CreateEvents(
+    wrapped_pipeline: Pipeline,
+    labeler: Labeler,
+    event_filter: EventFilter = NoFilter(),
+) -> _CreateEvents:
+    return _CreateEvents(wrapped_pipeline, _EventLabelWrapper(event_filter, labeler))
+
+
+class _CreateEvents(Composite):
```

Comment:
this one creates an internal Transformation out of the filter and the labeller, that it uses as the primary transformation. secondary transformation is the wrapped pipeline.
The composite in the end returns all timestamps, but NaNs where there's no event